test/08_faucet_tests.ts

```diff
@@ -60,48 +62,46 @@ describe('Testing the Holograph Faucet', async () => {
 
     it('isAllowedToWithdraw(): User is allowed to withdraw for the first time', async function () {
       expect(
-        await FAUCET.functions.isAllowedToWithdraw(DEPLOYER)
+        await FAUCET.isAllowedToWithdraw(DEPLOYER)
       ).to.be.true
     })
 
     it('requestTokens(): User can withdraw for the first time', async function () {
-      await FAUCET.functions.requestTokens().then(tx => tx.wait())
+      await FAUCET.requestTokens().then(tx => tx.wait())
       dripCount++
       expect(
-        await SAMPLEERC20.functions.balanceOf(DEPLOYER)
+        await ERC20.balanceOf(DEPLOYER)
       ).to.equal(DEFAULT_DRIP_AMOUNT)
     })
 
     it('isAllowedToWithdraw(): User is not allowed to withdraw for the second time', async function () {
       expect(
-        await FAUCET.functions.isAllowedToWithdraw(DEPLOYER)
+        await FAUCET.isAllowedToWithdraw(DEPLOYER)
       ).to.be.false
     })
 
     it('requestTokens(): User cannot withdraw for the second time', async function () {
-      expect(
-        await FAUCET.functions.requestTokens()
-      ).to.be.revertedWith(COME_BACK_LATER)
+      await expect(FAUCET.requestTokens()).to.be.revertedWith(COME_BACK_LATER)
```

Comment:
This is wrong. expect and its properties are not asynchronous (except `.resolves` an `.rejects`) https://jestjs.io/docs/asynchronous 


Correct would be
```
const result = await FAUCET.requestTokens()
expect(result).to.be.revertedWith(COME_BACK_LATER)
```

which is analogous to
```
expect(await FAUCET.requestTokens()).to.be.revertedWith(COME_BACK_LATER)
```

Reply:
Actually sorry, revertedWith is asynchronous!

Reply:
Await inside of the `expect` function is optional since it automatically handles promises. I do like the idea of still including it though, cuz it helps to better know how it's being run. And yes, `revertedWith` is asynchronous. As for breaking down the call and saving it as a `result` before using it in the `expect` call, that's your personal code preference. Feel free to do that, but please don't call it the "correct way", that is opinion.

Reply:
Interesting, from the docs and googling I understood that when you pass a promise in `expect()`, you need to use `resolves` or `rejects` ¯\\_(ツ)_/¯

Reply:
Oh wait a minute. We are not using Jest, but Chai?? That explains a lot